easybuild/easyconfigs/m/MPICH/MPICH-4.2.1-GCC-12.3.0.eb

```diff
@@ -0,0 +1,28 @@
+name = 'MPICH'
+version = '4.2.1'
+
+homepage = 'https://www.mpich.org/'
+description = """MPICH is a high-performance and widely portable implementation
+of the Message Passing Interface (MPI) standard (MPI-1, MPI-2 and MPI-3)."""
+
+toolchain = {'name': 'GCC', 'version': '12.3.0'}
+
+source_urls = ['https://www.mpich.org/static/downloads/%(version)s']
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['23331b2299f287c3419727edc2df8922d7e7abbb9fd0ac74e03b9966f9ad42d7']
+
+dependencies = [
+    ('UCX', '1.14.1'),
+    ('hwloc', '2.9.1'),
+    ('libfabric', '1.18.0'),
+]
+
+configopts = ' '.join([
+    '--with-device=ch4:ucx',
```

Comment:
```suggestion
dependencies = [
    ('UCX', '1.14.1'),
    ('hwloc', '2.9.1'),
    # include this (and comment out the UCX dependency) if you plan to build with ofi support
    # ('libfabric', '1.18.0'),
]

configopts = ' '.join([
    # MPICH can only be built for one backend communication  library at a time (it seems)
    # so you need to choose between UCX and libfabric at configuration time
    # (if using libfabric, as in the commented out line below, you will need to uncomment it as a dependency)
    # '--with-device=ch4:ofi',
    '--with-device=ch4:ucx',
```

Reply:
Thoughts on this @jfgrimm?

This seems like a reasonable default to me, since IB-based systems are way more prevalent than Omnipath...

Reply:
And add a `build_info_msg`, so that people know that this is the case.

Reply:
meant to revisit this
* leaving `libfabric` in as a dependency won't do any harm on IB systems, and makes it easier to adapt for OPX
* will add a `build_info_msg` (still planning to write some docs and point to that, also for OpenMPI)

Reply:
@jfgrimm What do you think of building it twice, one for each and then having an easyconfig parameter that chooses which to expose (could be a symlink with the installations themselves in subdirectories)? If we can force the symlink to be used when linking, changing to the alternate would just be updating the symlink.

Reply:
@ocaisa not a bad idea, I'll have a look at doing that